src/sdfg_diff_viewr.ts

```diff
@@ -0,0 +1,724 @@
+// Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+
+import $ from 'jquery';
+
+import {
+    DagreGraph,
+    graphFindRecursive,
+    htmlSanitize,
+    ISDFV,
+    ISDFVUserInterface,
+    JsonSDFG,
+    JsonSDFGControlFlowRegion,
+    JsonSDFGEdge,
+    JsonSDFGElement,
+    JsonSDFGState,
+    SDFG,
+    SDFGElement,
+    SDFGNode,
+    SDFVWebUI,
+    traverseSDFGScopes,
+    WebSDFV,
+} from '.';
+import { DiffOverlay } from './overlays/diff_overlay';
+import { SDFGRenderer } from './renderer/renderer';
+import _ from 'lodash';
+
+type ChangeState = 'nodiff' | 'changed' | 'added' | 'removed';
+
+type localGraphDiffStackEntry = {
+    guid: string | null,
+    htmlLabel: string,
+    changeStatus: ChangeState,
+    zoomToNodes: () => SDFGNode[],
+    indents: number,
+    children: (localGraphDiffStackEntry | null)[],
+};
+
+const DIFF_ORDER: ChangeState[] = ['nodiff', 'removed', 'added', 'changed'];
+
+const DIFF_IGNORE_ATTRIBUTES = [
+    'guid',
+    'hash',
+    'sdfg',
+    'orig_sdfg',
+    'transformation_hist',
+    'layout',
+];
+
+export interface DiffMap {
+    addedKeys: Set<string>;
+    removedKeys: Set<string>;
+    changedKeys: Set<string>;
+}
+
+export abstract class SDFGDiffViewer implements ISDFV {
+
+    protected diffMap?: DiffMap;
+
+    public constructor(
+        protected readonly leftRenderer: SDFGRenderer,
+        protected readonly rightRenderer: SDFGRenderer
+    ) {
+    }
+
+    protected destroy(): void {
+        this.leftRenderer.destroy();
+        this.rightRenderer.destroy();
+    }
+
+    public static async diff(
+        graphA: JsonSDFG, graphB: JsonSDFG
+    ): Promise<DiffMap> {
+        if (!graphA.attributes.guid || !graphB.attributes.guid) {
```

Comment:
I would use the SDFGElement's guid() method below, and when diff is called on graphs without guid here (i.e., this branch should not change) I would alert the user that diffs are not supported for graphs older than version 0.16.2 (?)

Reply:
That also means we have to add guid() to JsonSDFGElement? :/

Reply:
Found a way around it

Reply:
Added a warning